src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -411,20 +409,26 @@ SafeToPushdownWindowFunction(Query *query, StringInfo *errorDetail)
 
 		if (!windowClause->partitionClause)
 		{
-			*errorDetail = makeStringInfo();
-			appendStringInfoString(*errorDetail,
-								   "Window functions without PARTITION BY on distribution "
-								   "column is currently unsupported");
+			if (errorDetail)
+			{
+				*errorDetail = makeStringInfo();
+				appendStringInfoString(*errorDetail,
+									   "Window functions without PARTITION BY on distribution "
+									   "column is currently unsupported");
```

Comment:
is this error message still valid anywhere after we land this patch?

Reply:
`multi_insert_select_window` still emits it as a reason to not push down a subquery